### ISSUE TITLE
Custom Loss functions for MPS

### ISSUE THREAD

**Issue by MartinLNG** [MartinLNG/adversarial_training_mps#45]
MPS interpreted as Born machine outputs not the type of logit torch.nn loss functions expect (pre-softmax basically). To have efficient loss computation, custom loss functions would be useful.
